inst/plumber/v1/endpoints.R

```diff
@@ -406,6 +421,77 @@ function(req) {
   lkups$versions_paths[[req$argsQuery$version]]$cache_data_id
 }
 
+#* Retrieve grouped data stats
+#* @get /api/v1/grouped-stats
+#* @param welfare:[dbl] numeric vector for welfare
+#* @param population:[dbl] numeric vector for population
+#* @param requested_mean:[dbl] mean value
+#* @param povline:[dbl] poverty line value
+#* @param format:[chr] Response format. Options are "json", "csv", "rds", or "arrow".
+function(req, res) {
+  ### TO DO :
+  # Working with args instead of argsQuery because we do not have type and valid values in lkup$query_controls
+  # We need to change `lkup` to have valid_values and type added
+  # Talk with Andres/Tony on doing it.
+  params <- req$args
+  relevant_params <- params[names(params) != "format"]
+  relevant_params <- lapply(relevant_params, as.numeric)
+  res$serializer <- pipapi::assign_serializer(format = params$format)
+  out <- do.call(wbpip:::gd_compute_pip_stats, relevant_params)
+  if(!is.null(params$format) && params$format == "csv") {
+    out <- change_grouped_stats_to_csv(out)
+  }
+  out
+}
+
+#* Retrieve regression parameters
+#* @get /api/v1/regression-params
+#* @param welfare:[dbl] numeric vector for welfare
+#* @param population:[dbl] numeric vector for population
+#* @param format:[chr] Response format. Options are "json", "csv", "rds", or "arrow".
+function(req, res) {
+  ### TO DO :
+  # Working with args instead of argsQuery because we do not have type and valid values in lkup$query_controls
+  # We need to change `lkup` to have valid_values and type added
+  # Talk with Andres/Tony on doing it.
+  params <- req$args
+  params$weight <- params$population
+  params$population <- NULL
+  relevant_params <- params[names(params) != "format"]
+  res$serializer <- pipapi::assign_serializer(format = params$format)

```

Comment:
I think this logic could be simpler if the serializer is assigned at the beginning, no? Similar to what is done for the `pip-grp` endpoint for instance